File: Cerebro Utilities/Models/CardEntity.cs
using Azure.Search.Documents.Indexes;
using Microsoft.Azure.Cosmos.Table;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Cerebro_Utilities.Models
{
    public class CardEntity : TableEntity
    {
        public const string INDEX_NAME = "cerebrocards-index";

        public const string TABLE_NAME = "CerebroCards";

        public CardEntity() { }

        [SimpleField]
        public string Acceleration { get; set; }

        [SimpleField]
        public string Attack { get; set; }

        [SimpleField]
        public string Boost { get; set; }

        [SimpleField]
        public string Classification { get; set; }

        [SimpleField]
        public string Cost { get; set; }
        
        [SimpleField]
        public string Defense { get; set; }
        
        [SimpleField]
        public string Flavor { get; set; }

        [SimpleField]
        public string Group { get; set; }

        [SimpleField]
        public string Hand { get; set; }
        
        [SimpleField]
        public string Health { get; set; }

        [SimpleField(IsFilterable = true)]
        public bool Incomplete { get; set; }

        [SearchableField(IsFilterable = true, IsSortable = true)]
        public string Name { get; set; }

        [JsonIgnore]
        public List<PrintingEntity> Printings { get; set; }
        
        [SimpleField]
        public string Recover { get; set; }
        
        [SimpleField]
        public string Resource { get; set; }

        [SearchableField(IsFilterable = true)]
        public string Rules { get; set; }
        
        [SimpleField]
        public string Scheme { get; set; }
        
        [SimpleField]
        public bool Slash { get; set; }

        [SearchableField(IsFilterable = true)]
        public string Special { get; set; }

        [SearchableField(IsFilterable = true)]
        public string Stage { get; set; }

        [SearchableField(IsFilterable = true, IsSortable = true)]
        public string Subname { get; set; }
        
        [SimpleField]
        public string Threat { get; set; }
        
        [SimpleField]
        public string Threshold { get; set; }
        
        [SimpleField]
        public string Thwart { get; set; }

        [SearchableField(IsFilterable = true)]
        public string Traits { get; set; }

        [SearchableField(IsFilterable = true, IsSortable = true)]
        public string Type { get; set; }
        
        [SimpleField]
        public bool Unique { get; set; }
    }
}

File: models/cardEntity.js
const { BaseAuthoredEntity } = require('./baseAuthoredEntity');

class CardEntity extends BaseAuthoredEntity {
    static DATABASE_SUFFIX = 'cards';
    static COLLECTION = 'cards';

    constructor (document) {
        super(document);

        this.Acceleration = document.Acceleration;
        this.ArtificialPackId = document.ArtificialPackId;
        this.Artists = document.Artists;
        this.Attack = document.Attack;
        this.Boost = document.Boost;
        this.Classification = document.Classification;
        this.Cost = document.Cost;
        this.Defense = document.Defense;
        this.Flavor = document.Flavor;
        this.GroupId = document.GroupId;
        this.Hand = document.Hand;
        this.Health = document.Health;
        this.Incomplete = document.Incomplete;
        this.Name = document.Name;
        this.Printings = document.Printings;
        this.Queries = document.Queries;
        this.Recover = document.Recover;
        this.Resource = document.Resource;
        this.Rules = document.Rules;
        this.Scheme = document.Scheme;
        this.Slash = document.Slash;
        this.Special = document.Special;
        this.SpoilerTag = document.SpoilerTag;
        this.Stage = document.Stage;
        this.StartingThreat = document.StartingThreat;
        this.Subname = document.Subname;
        this.TargetThreat = document.TargetThreat;
        this.Thwart = document.Thwart;
        this.Traits = document.Traits;
        this.Type = document.Type;
        this.Unique = document.Unique;
    }
}

module.exports = { CardEntity }